doc/experiment.rst

```diff
@@ -1,327 +1,423 @@
+within an Experiment. In the following subsections, we define the
+general purpose of the three entities that can be created via
+Experiment API factory methods:
+
+* ``Orchestrator``
+* ``Model``
+* ``Ensemble``
+
+To create a reference to each entity object, use the associated
+``Experiment.create_...()`` function.
+
+.. list-table:: Experiment API Entity Creation
+   :widths: 20 65 25
+   :header-rows: 1
+
+   * - Factory Method
+     - Example
+     - Return Type
+   * - ``create_database()``
+     - ``orch = exp.create_database([port, db_nodes, ...])``
+     - :ref:`Orchestrator <orc_api>`
+   * - ``create_model()``
+     - ``model = exp.create_model(name, run_settings)``
+     - :ref:`Model <mode_api>`
+   * - ``create_ensemble()``
+     - ``ensemble = exp.create_ensemble(name[, params, ...])``
+     - :ref:`Ensemble <ensem_api>`
+
+Each entity instance can be used to start,
+monitor, and stop simulations from the notebook
+using the :ref:`Experiment API<experiment_api>`.
+
+.. list-table:: Interact with Entities during the Experiment
+   :widths: 25 55 25
+   :header-rows: 1
+
+   * - Experiment Function
+     - Example
+     - Desc
+   * - ``start()``
+     - ``exp.start(*args[, block, summary, ...])``
+     - Launch an Entity
+   * - ``stop()``
+     - ``exp.stop(*args)``
+     - Clobber an Entity
+   * - ``get_status()``
+     - ``exp.get_status(*args)``
+     - Retrieve Entity Status
+
+Orchestrator
+------------
+The orchestrator is an in-memory database with features designed
+to enable a wide variety of AI-enabled workflows, including features
+for online training, low-latency inference, cross-application data
+exchange, online interactive visualization, online data analysis, computational
+steering, and more. The ``Orchestrator`` can be thought of as a general
+feature store capable of storing numerical data, ML models, and scripts
+and capable of performing inference and script evaluation on feature
+store data. Any SmartSim ``Model`` or ``Ensemble`` model can connect to the
+``Orchestrator`` via the :ref:`SmartRedis<SmartRedis Client Library Hook>`
+client library to transmit data, execute ML models, and execute scripts.
+
+**SmartSim offers two types Orchestrator deployments:**
+
+* :ref:`Clustered Orchestrator <Clustered Orchestrator>`
+* :ref:`Colocated Orchestrator <Colocated Orchestrator>`
+
+Clustered Orchestrator
+^^^^^^^^^^^^^^^^^^^^^^
+The orchestrator can be composed of one or more in-memory database shards that are spread
+across one or more compute nodes.
+The multiple compute hosts memory can be used together to store data.
+Users do not need to know how the data is stored in a clustered
+configuration and can address the cluster with a SmartRedis client
+like a single block of memory using simple put/get semantics in SmartRedis.
+The database shards communicate with each other via TCP/IP in the driver script and application.
+SmartRedis will ensure that data is evenly distributed among all nodes in the cluster.
```

Comment:
This last sentence is inaccurate. SmartRedis does only a minimal attempt at load balancing. Mostly it's up to Redis.

Reply:
It's not even really much up to Redis since we directly address the shards when we send the data. We rely on the fact that CRC16 uniformly, but randomly calculates hashes to ensure that the number of keys on each shard is roughly the same. This doesn't prevent the case where some of the keys might be storing larger amounts of data though.